packages/gestalt/src/Tooltip/InternalTooltip.js

```diff
@@ -0,0 +1,168 @@
+// @flow strict
+import { type Node, useReducer, useRef } from 'react';
+import Box from '../Box.js';
+import Controller from '../Controller.js';
+import Layer from '../Layer.js';
+import Text from '../Text.js';
+import useDebouncedCallback from '../useDebouncedCallback.js';
+import { type Indexable } from '../zIndex.js';
+
+const noop = () => {};
+const TIMEOUT = 100;
+
+const initialState = { hoveredIcon: false, hoveredText: false, isOpen: false };
+
+const reducer = (
+  state: {| hoveredIcon: boolean, hoveredText: boolean, isOpen: boolean |},
+  action: {| type: 'hoverInIcon' | 'hoverInText' | 'hoverOutIcon' | 'hoverOutText' |},
+) => {
+  switch (action.type) {
+    case 'hoverInIcon':
+      return {
+        ...state,
+        hoveredIcon: true,
+        isOpen: true,
+      };
+    case 'hoverInText':
+      return {
+        ...state,
+        hoveredText: true,
+        isOpen: true,
+      };
+    case 'hoverOutIcon':
+      return {
+        ...state,
+        hoveredIcon: false,
+        isOpen: !state.hoveredText ? false : state.isOpen,
+      };
+    case 'hoverOutText':
+      return {
+        ...state,
+        hoveredText: false,
+        isOpen: !state.hoveredIcon ? false : state.isOpen,
+      };
+    default:
+      throw new Error();
+  }
+};
+
+type Props = {|
+  /**
```

Comment:
can we remove all prop descriptions? Duplicated from tooltip.

I only keep external API props descriptions